fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/PrestoSqlBackedClient.java

```diff
@@ -223,20 +223,30 @@ protected static String sqlQueryToPrestoQuery(String sqlQuery) {
         String fieldValue;
         String filterClause;
         String filterClauseFixed;
-        int equalSignIndex;
+        int equalIndex;
+        int notEqualIndex;
+        int comparatorIndex;
+        // could be either <> or =
+        String comparator;
         for (int i = 2; i < filterClauses.length; i++) {
             filterClause = filterClauses[i];
-            if (filterClauses[i].contains("=")) {
-                equalSignIndex = filterClause.indexOf("=");
-                fieldName = filterClause.substring(0, equalSignIndex).trim();
-                fieldValue = filterClause.substring(equalSignIndex + 1).trim();
-                filterClauseFixed = String.format("CAST(%s AS varchar) = %s", fieldName, fieldValue);
-                fixFilterPrestoQuery = fixFilterPrestoQuery.replace(filterClause, filterClauseFixed);
+            equalIndex = filterClause.indexOf("=");
+            notEqualIndex = filterClause.indexOf("<>");
+            if (equalIndex == -1 && notEqualIndex == -1) {
+                continue;
+            } else if (equalIndex == -1 || (notEqualIndex != -1 && notEqualIndex < equalIndex)) {
```

Comment:
I feel like this is a pretty confusing conditional for what it's doing, maybe rewrite it to be something like

```java
if (notEqualIndex != -1 && (equalIndex == -1 || notEqualIndex < equalIndex)) {
 // not equal
} else if (equalIndex != -1 && (notEqualIndex == -1 || equalIndex < notEqualIndex)) {
  // equal
}
```

Reply:
sure. that does seems to be cleaner. I think we do need to catch the case where both indexes are `-1`. This makes sure the function works when we add other filter clauses other than `=` and `<>` in the further.
```java
if (notEqualIndex != -1 && (equalIndex == -1 || notEqualIndex < equalIndex)) {
  // not equal
} else if (equalIndex != -1 && (notEqualIndex == -1 || equalIndex < notEqualIndex)) {
  // equal
} else {
 // do nothing jump to the next expression
}
```